### DISCUSSION TITLE
How do you test your configuration on a bare OS? Docker? Distrobox? Vagrant?

### DISCUSSION THREAD

**Discussion by stephane-klein** [twpayne/chezmoi#3695] [Q&A] [answered]
Hello,

Can you recommend a particular method to easily test my Chezmoi configuration from a blank OS?

I'm using Fedora 39 on my workstation.

Several ideas come to mind:

- a. Test in a Docker container with a Fedora image
- b. Test with [Distrobox](https://distrobox.it/), in a blank home - but I'm afraid of doing something stupid and for some reason modifying the real home of my workstation.
- c. Test with Virtualbox or directly with Qemu

Currently, I'm thinking of choosing the solution `a`.

I'm curious, what is your method?

Best regards,  
Stéphane

**Comment by stephane-klein** [comment]
[Here](https://github.com/shunk031/dotfiles?tab=readme-ov-file#%EF%B8%8F-update--test-) I read:

> To verify that the updated scripts work correctly, run the scripts on the actual local machine and on the docker container

**Comment by twpayne** [comment, marked as answer]
Docker is probably the easiest:

```console
$ docker run -it fedora:latest
# sh -c "$(curl -fsLS get.chezmoi.io)" -- init --apply $GITHUB_USERNAME
```
